lib/matplotlib/tests/test_figure.py

```diff
@@ -967,6 +967,39 @@ def test_subfigure_double():
     axsRight = subfigs[1].subplots(2, 2)
 
 
+def test_subfigure_spanning():
+    # test that subfigures get laid out properly...
+    fig = plt.figure(constrained_layout=True)
+    gs = fig.add_gridspec(3, 3)
+    sub_figs = []
+    sub_figs += [fig.add_subfigure(gs[0, 0])]
+    sub_figs += [fig.add_subfigure(gs[0:2, 1])]
+    sub_figs += [fig.add_subfigure(gs[2, 1:3])]
+
+    w = 640
+    h = 480
+    minp = sub_figs[0].bbox.min
+    exp = np.array([0., h*2/3])
+    np.testing.assert_allclose(minp, exp)
+    maxp = sub_figs[0].bbox.max
+    exp = np.array([w / 3, h])
+    np.testing.assert_allclose(maxp, exp)
```

Comment:
```suggestion
    np.testing.assert_allclose(sub_figs[0].bbox.max, [w / 3, h])
```
is more concise compared to indirecting to local variables.

If that feels to condensed, you can alternatively do:
```
    np.testing.assert_allclose(
        sub_figs[0].bbox.max,
        [w / 3, h])
```

This applies to various assertions.

Reply:
I think this was to be applied in all cases.

Reply:
done!